R/add_transition_risk_category_at_company_level.R

```diff
@@ -0,0 +1,14 @@
+add_transition_risk_category_at_company_level <- function(data) {
+  data |>
+    rename(grouped_by = "benchmark_tr_score", risk_category = "transition_risk_category") |>
+    epa_at_company_level() |>
+    insert_row_with_na_in_risk_category() |>
+    rename(benchmark_tr_score_avg = "grouped_by", transition_risk_category = "risk_category", transition_risk_category_share = "value")
+}
+
+join_risk_categories_at_company_level <- function(data, risk_categories) {
+  data |>
+    mutate(original_emission_profile = .data$emission_profile) |>
+    left_join(risk_categories, by = c("companies_id", "benchmark_tr_score_avg", "emission_profile" = "transition_risk_category")) |>
+    rename(transition_risk_category = "original_emission_profile")
+}
```

Comment:
@maurolepore I haven't shifted this function to `tiltTransitionRisk` because we don't have the company level dataset of transition risk profile in `tiltTransitionRisk`, however I would be happy to shift these functions to `tiltTransitionRisk` if you think otherwise. :)

FYI: We could create a separate ticket which will shift all the functions related to tranistion risk profile indicator involving `score_transition_risk` and `score_transition_risk_polish` to the repo `tiltTransitionRisk`.

Reply:
I fail to follow the argument but I think it's fine. The architecture is structured to serve your purposes. Use it however it makes your work easier to do and maintain in the long term.

While I don't quite understand the comment, my intuition suggests that if you want to move a function to a higher-level component (tiltTransitionRisk) but you can't because the function mentions lower-level concepts (column names) that don't exist in the higher-level component, then likely this means that the function is mixing code at different levels of abstraction that may belong to separate functions. Above I noted some code that may move to an adaptor and that may solve the problem.

Reply:
Adaptors won't solve this issue. This issue can be solved if all the lower level functions are moved to the higher level component. Should I create a separate ticket for this work?

Reply:
Ticket: https://github.com/2DegreesInvesting/tiltIndicatorAfter/issues/268